add windows build

dogboydog opened the pull request:
Not working right now, but it's a start.  There's no clear error message I can see in the output

Maybe the install step needs some adjustment on windows?

Closes #5

dogboydog committed:
add windows build

bjorn commented:
The error on the Windows build is:
```
2024-12-28T01:16:41.1617961Z   image_io.cpp
2024-12-28T01:16:41.6068343Z D:\a\qaseprite\qaseprite\aseprite\third_party\zlib\zlib.h(34,10): error C1083: Cannot open include file: 'zconf.h': No such file or directory [D:\a\qaseprite\qaseprite\build\aseprite\src\doc\doc-lib.vcxproj]
2024-12-28T01:16:41.6069671Z   (compiling source file '../../../../aseprite/src/doc/image_io.cpp')
```
Which might be related to these earlier messages:
```
2024-12-28T01:12:13.9906174Z -- Renaming
2024-12-28T01:12:13.9906587Z --     D:/a/qaseprite/qaseprite/aseprite/third_party/zlib/zconf.h
2024-12-28T01:12:13.9907222Z -- to 'zconf.h.included' because this file is included with zlib
2024-12-28T01:12:13.9907809Z -- but CMake generates it automatically in the build directory.
```
Seems like either CMake didn't generate the `zconf.h` after all, or the location at which it was generated wasn't added to the include path.

bjorn committed:
A bit of cleanup

Removed some unused variables

bjorn committed:
Try to build against MinGW Qt

bjorn committed:
Try using shared ZLIB

Also tried a shorter way of setting the compilers.

bjorn committed:
Try applying patch to fix zlib include dir

Also raised minimum CMake version to match the one of Aseprite.

bjorn committed:
Using CC/CXX env vars for setting compiler didn't appear to work

bjorn committed:
Try a VS build first, since it's failing to use MinGW

bjorn committed:
Apply patches sync RuntimeLibrary with the one used by Qt

bjorn committed:
Removed ccache setup on Windows build because it's not used for now

CCache does claim support for MSVC at
https://ccache.dev/platform-compiler-language-support.html but the build
doesn't seem to use it automatically like on Linux and macOS.

bjorn committed:
Try to fix RuntimeLibrary selection

That's what you get from blindly trying suggestions by AI...

bjorn commented:
Alright, at least the MSVC build succeeded, but the install location could be improved:
```
-- Installing: C:/Program Files (x86)/qaseprite/lib/qaseprite.lib
-- Installing: C:/Program Files (x86)/qaseprite/bin/qaseprite.dll
```
In our build environment, something is setting a `QT_PLUGIN_PATH` environment variable, which we could use. I'm not sure why the `${QT_INSTALL_PREFIX}/${QT_INSTALL_PLUGINS}` thing doesn't seem to be working on Windows.

bjorn committed:
Try to fix install issues

* Set RUNTIME DESTINATION, which is used for DLLs
* Use MODULE instead of SHARED, to avoid creating (and installing) a
  needless .lib file on Windows.

bjorn commented:
Whoops, after merging this I noticed the "32-bit" build is actually also a 64-bit build. I should have used `win32_msvc2019` and not `win64_msvc2019_64` when installing Qt 5.15.2 (fixed in #7).
